### PR TITLE
Translate Nelson log messages, fix and add a --dry-run option

### PR DESCRIPTION
**Pull request by pierallard**
- Translates the log message
- Fix the fact that files can not be alone in the main folder
- Add a --dry-run option

### CODE REVIEW

File: src/Akeneo/System/AbstractConsoleLogger.php [hunk 1]
```diff
@@ -43,30 +49,64 @@ protected function getTime()
     /**
      * Write a processing message
      *
-     * @param string $comment
+     * @param string   $message
+     * @param string[] $messageParams
      */
-    protected function writeProcessing($comment)
+    protected function writeProcessing($message, $messageParams = [])
```
Comment (nidup) [inline src/Akeneo/System/AbstractConsoleLogger.php]: :+1:

File: src/Akeneo/System/AbstractConsoleLogger.php [hunk 2]
```diff
@@ -43,30 +49,62 @@ protected function getTime()
...
     {
-        $this->output->writeln(sprintf('%s <comment>%s<blink>...</blink></comment>', $this->getTime(), $comment));
+        $this->output->writeln(sprintf(
+            '%s <comment>%s<blink>...</blink></comment>',
+            $this->getTime(),
+            $this->translator->trans($message, $messageParams)
```
Comment (nidup) [inline src/Akeneo/System/AbstractConsoleLogger.php]: :tada:

File: src/Akeneo/System/AbstractConsoleLogger.php [hunk 3]
```diff
@@ -43,30 +49,62 @@ protected function getTime()
...
+
+    /**
+     * Surround translation keys with '%' and bold values
+     *
+     * @param $params
+     *
+     * @return array
      */
-    protected function writeSuccess($success)
+    protected function getTranslationParams($params)
```
Comment (nidup) [inline src/Akeneo/System/AbstractConsoleLogger.php]: getTranslationParams -> prepareTranslationParams ?

File: src/Akeneo/Nelson/ConsoleLogger.php [hunk 4]
```diff
@@ -34,36 +34,30 @@ public static function getSubscribedEvents()
...
     /**
      * @param GenericEvent $event
      */
     public function preNelsonPush(GenericEvent $event)
     {
-        $this->writeProcessing(sprintf(
-            'Pushing translation keys of branch <bold>%s</bold> from Github to Crowdin',
-            $event->getArgument('branch')
-        ));
+        $this->writeProcessing(Events::PRE_NELSON_PUSH, $this->getTranslationParams($event->getArguments()));
```
Comment (nidup) [inline src/Akeneo/Nelson/ConsoleLogger.php]: getTranslationParams could be done inside the method writeProcessing to group the "translation concern" in a single place

File: src/Akeneo/Git/PullRequestCreator.php [hunk 5]
```diff
@@ -64,41 +64,45 @@ public function __construct(
...
+            'branch' => $baseBranch,
+            'dryRun' => $dryRun,
         ]));
 
-        $this->executor->execute(sprintf('cd %s && git checkout -B crowdin/%s', $projectDir, $branch));
+        if (!$dryRun) {
+            $this->executor->execute(sprintf('cd %s && git checkout -B crowdin/%s', $projectDir, $branch));
 
-        $this->executor->execute(sprintf('cd %s && git add .', $projectDir));
+            $this->executor->execute(sprintf('cd %s && git add .', $projectDir));
```
Comment (nidup) [inline src/Akeneo/Git/PullRequestCreator.php]: broken align

Comment (pierallard) [inline src/Akeneo/Git/PullRequestCreator.php]: Normal, I added a condition.

**Comment by nidup** [comment]
gtm

**Pull request merged**